### DISCUSSION TITLE
ModuleNotFoundError: No module named 'librosa'

### DISCUSSION THREAD

**Discussion by LeromX** [mcguirepr89/BirdNET-Pi#839] [Q&A] [answered]
When trying to make BirdNET-Pi run on a Raspberry Pi 3B that i had lying around I've run in to this message looking in the Log:
19:26:00---Traceback (most recent call last):
19:26:00---  File "/usr/local/bin/server.py", line 11, in <module>
19:26:00---    import librosa
19:26:00---ModuleNotFoundError: No module named 'librosa'

Anyone have any idea?

I´ve activated memory swap cuz it seems to be a requirement.

**Comment by ehpersonal38** [comment, marked as answer]
BirdNET-Pi uses Python under a virtual environment, so you'll have to execute the following command to enter that environment and install the required packages. This should've been done during installation but seemed to have failed on your system for some reason?

```
source ~/BirdNET-Pi/birdnet/bin/activate &&
pip3 install -U -r ~/BirdNET-Pi/requirements.txt &&
deactivate
```

Then after a reboot or `bash ~/BirdNET-Pi/scripts/restart_services.sh`, hopefully the issue will be fixed!

**Reply by LeromX** [reply]
Thank You soo much. I successfully run the first line of commands and i've rebooted the Pi.
Now i can see these messages in the log:
08:29:33---ImportError: cannot import name 'markdown' from 'markdown' (BirdNET-Pi/birdnet/lib/python3.9/site-packages/markdown/__init__.py)
08:29:53---Traceback (most recent call last):
08:29:53---  File "/usr/local/bin/server.py", line 18, in <module>
08:29:53---    from utils.notifications import sendAppriseNotifications
08:29:53---  File "BirdNET-Pi/scripts/utils/notifications.py", line 1, in <module>
08:29:53---    import apprise
08:29:53---  File "BirdNET-Pi/birdnet/lib/python3.9/site-packages/apprise/__init__.py", line 55, in <module>
08:29:53---    from .Apprise import Apprise
08:29:53---  File "BirdNET-Pi/birdnet/lib/python3.9/site-packages/apprise/Apprise.py", line 29, in <module>
08:29:53---    from .conversion import convert_between
08:29:53---  File "BirdNET-Pi/birdnet/lib/python3.9/site-packages/apprise/conversion.py", line 27, in <module>
08:29:53---    from markdown import markdown

BR
Henrik

**Comment by elaheJ** [comment]
this still does not work for me:
source ~/BirdNET-Pi/birdnet/bin/activate &&
pip3 install -U -r ~/BirdNET-Pi/requirements.txt &&
deactivate

librosa is installed here: /usr/local/bin$ /usr/local/lib/python3.9/dist-packages/

**Reply by ehpersonal38** [reply]
I'd recommend a fresh reimage and install
